tests/integration/models/device.spec.coffee

```diff
@@ -1028,6 +1028,18 @@ describe 'Device Model', ->
 					m.chai.expect(_.find(result, { name: 'A' }).value).equal('a')
 					m.chai.expect(_.find(result, { name: 'B' }).value).equal('b')
 
+		# TODO: xginn8: left an `only` here for testing
+		describe.only 'balena.models.device.getSupervisorTargetState()', ->
+
+			it 'should be rejected if the device does not exist', ->
+				promise = balena.models.device.getSupervisorTargetState('asdfghjkl')
+				m.chai.expect(promise).to.be.rejectedWith('Device not found: asdfghjkl')
+
+			it 'should reflect the device\'s state', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) =>
+					m.chai.expect(state.name).to.equal(@device.name)
+					m.chai.expect(state.apps[@application.id].name).to.equal(@application.name)
```

Comment:
It would be nice to also include some extra `.to.be.a('number')` or `.to.be.a('string')` checks, that will more or less check what the typings describe about this object.
This way we will notice if suddenly the state endpoint response format changes.

Reply:
@thgreasi i cannot figure out why `@application` isn't defined in https://github.com/balena-io/balena-sdk/pull/653/files#diff-499041c13cee047a446c4f6ee3a36c14R1045, yet seems to be defined in the above definition (https://github.com/balena-io/balena-sdk/pull/653/files#diff-499041c13cee047a446c4f6ee3a36c14R1027)